lib/ferrum/contexts.rb

```diff
@@ -20,6 +21,20 @@ def default_context
       @default_context ||= create
     end
 
+    def create_default_context
+      default_context_id = compute_default_context_id
+      # Targets created in this context will not be created with a browserContextId
+      @contexts[default_context_id] = ::Ferrum::Context.new(@client, self, nil)
+    end
+
+    # Compute the default context ID by looking for contexts not returned by Target.getBrowserContexts
```

Comment:
Hm this method was also suggested (independently) by one of the Chromium engineers, so it might be fine. I am slightly nervous about the case when no tabs exist in the default context which presumably could happen, but I think generally people using browser automation should be able to ensure that the correct prerequisites are met.

One other approach might be to have the `Contexts` class (optionally) enumerate all existing contexts at startup and create entries for them, and then any targets that aren't part of the discovered contexts are probably from the default context. This could get out of sync potentially, but I'm not sure if there is really any problem with a Target (in Ruby) being assigned to the incorrect Context object from the Contexts map, since the Target seems to have sufficient information to operate on its own and doesn't really need info from the Context. I think the main purpose of that association is that when `Context#dispose` is called, it closes the connections to the `Targets` in that `Context`, so that's probably not going to be an issue in practice because if Ferrum is not aware of that `Context` you're never going to call `#dispose` on it.